### ISSUE TITLE
Make Training Work

### ISSUE THREAD

**Issue by Bonifatius94** [Bonifatius94/Schafkopf.AI#5]

**Comment by Bonifatius94** [comment]
- Put neural net framework in place
  - [x] can solve regression task (e.g. predict sin(x))
  - [x] can solve classification task  (e.g. learn MNIST)
- Implement RL algorithms
  - [ ] Deep Q
  - [x] PPO
  - [ ] Monte Carlo Tree Search
- Prepare training data based on gameplay
  - [x] model a reward function
  - [x] model the input features seen by the agent
- Implement Benchmark
  - [x] random play benchmark
  - [ ] benchmark against heuristic agent
  - [ ] benchmark against best trained agent
  - [ ] benchmark with real gameplay data
